packages/core/src/amazonqGumby/chat/views/actions/uiMessageListener.ts

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { MessageListener } from '../../../../amazonq/messages/messageListener'
+import { ExtensionMessage } from '../../../../amazonq/webview/ui/commands'
+import { ChatControllerEventEmitters } from '../../controller/controller'
+
+export interface UIMessageListenerProps {
+    readonly chatControllerEventEmitters: ChatControllerEventEmitters
+    readonly webViewMessageListener: MessageListener<any>
+}
+
+export class UIMessageListener {
```

Comment:
This is another thing that was copy-pasted from featureDev? https://github.com/aws/aws-toolkit-vscode/blob/46560e70b7dd29d83ff557b1db246db5091dd7fe/packages/core/src/amazonqFeatureDev/views/actions/uiMessageListener.ts#L15

2000 lines of code. It's very hard to review PRs when so much code is copy-pasted. And then it's hard in the future to maintain the project.

Reply:
Like I said in the previous comment, I'll see if I can create a base class that both FeatureDev and Gumby can inherit from. This is one of those issues where FeatureDev needs to build a yellow house with 3 walls and 2 railings, and Gumby needs to build a red house with a garage and 6 railings, so there is going to be very similar looking code regardless.